docs/guide.md

```diff
@@ -0,0 +1,761 @@
+# Guide to Running E3SM
```

Comment:
We can name each sub-guide accordingly. E.g., "Guide to Running E3SM - Production"